File: examples/Phonebook/ocaml-server/Phonebook/major_0.ml

////////////////////////////////////////////////////////
// Types
////////////////////////////////////////////////////////

// Struct: address
type address = {
  street : street;
  city : city;
  zipcode : zipcode;
  state : state;
}

// Struct: person
type person = {
  name : name;
  home_number : phone;
  cell_number : phone;
  address : address maybe;
  friends : (person_id list);
}

// Struct: lookup_person
type lookup_person = {
  id : person_id;
}

// Struct: lookup_person_by_name
type lookup_person_by_name = {
  name : name;
}

// Struct: insert_person
type insert_person = {
  person : person;
}

File: examples/Phonebook/go-server/phonebook/major_0.go

////////////////////////////////////////////////////////
// Types
////////////////////////////////////////////////////////


// Struct: Address
type Address struct {
    street Street
    city City
    zipcode Zipcode
    state State
}

// Struct: Person
type Person struct {
    name Name
    home_number Phone
    cell_number Phone
    address Address*
    friends List
}

// Struct: LookupPerson
type LookupPerson struct {
    id PersonId
}

// Struct: LookupPersonByName
type LookupPersonByName struct {
    name Name
}

// Struct: InsertPerson
type InsertPerson struct {
    person Person
}
